python/ray/data/dataset.py

```diff
@@ -1060,7 +1060,13 @@ def filter(
         return Dataset(plan, logical_plan)
 
     @AllToAllAPI
-    def repartition(self, num_blocks: int, *, shuffle: bool = False) -> "Dataset":
+    def repartition(
+        self,
+        num_blocks: int,
+        *,
+        shuffle: bool = False,
+        _debug_limit_shuffle_execution_to_num_blocks: Optional[int] = None,
```

Comment:
I don't think we want to put the debugging parameter into API. It would be confusing for users even though it starts with `_`.

Given this is only for release test, can we just add a new config to `DataContext`, and check the config in `pull_based_shuffle_task_scheduler.py`? This also avoids passing the parameter through `LogicalOperator`.

In the future, if we do see the need to config `_debug_limit_shuffle_execution_to_num_blocks` per operator, then we can figure out another way.

Reply:
+1, using `DataContext` or env vars would be better

Reply:
Hmm I actually intended for this to be user-facing, so that you can more easily debug large-scale shuffles. I also slightly prefer this way of setting the config because it is clear that it only applies to this Dataset call, and it is obvious when reading the docs page how to do it. We sometimes use the `_` naming in Ray Core PublicAPIs as well, for power user features. It does need a docstring explaining usage though; I haven't added this yet.

But I am okay to switch to DataContext if you feel strongly about it.

Reply:
Got it. How about we do the `DataContext` config thing first, and test with real user. If they found useful, then we add it to the API?